src/stores/checkinStore/index.js

```diff
@@ -62,26 +112,65 @@ class CheckinStore {
     )
   }
 
-  @action handleCheckin = (history) => {
+
+  @action async handleCheckin(history) {
+    console.log(history)
+
+    //get current block number and default address
+    const blockNumber = await appchain.base.getBlockNumber()
+    const defaultAccount = await appchain.base.getDefaultAccount()
+
+    //get control contract addr
+    const clubAddr = this.selectedClubAddr[0]
+    const clubDataContract = new appchain.base.Contract(dataAbi, clubAddr)
+    const clubControlAddr = await clubDataContract.methods.controlAddress().call()
+
+    //get current text
+    const checkInText = this.checkinContent
+    let checkInImg = 'NA'
+    
+    //construct transaction
+    const tx = {
+      ...transaction,
+      from: defaultAccount,
+      validUntilBlock: blockNumber + 88,
+    }
+
     if (this.files.length) {
-      log('有图')
-      handleUploadImage(this.files)
-        .then(res => {
-          log('ok, ', res)
-          if (res.hash) {
-            log('pic name', res.key)
-            log(timeConverter(Math.round(new Date().getTime() / 1000)))
-            //TODO interact with chain
-          }
-        })
-        .catch((err) => {
-          this.handleCheckinFailed(err.code)
-        })
+      checkInImg = await handleUploadImage(this.files)
+      checkInImg = checkInImg['key']
+      console.log('checkinimg',checkInImg)
+    }
+
+    const clubControlContract = new appchain.base.Contract(controlAbi, clubControlAddr)
+    const txHash = await clubControlContract.methods.checkin(checkInImg, checkInText).send(tx)            
+    const receipt = await appchain.listeners.listenToTransactionReceipt(txHash.hash)
+
+    console.log('receipt', receipt)
+
+    if (receipt.errorMessage === null) {
+      log('checkin success')
```

Comment:
if check in successed, you should call `handleCheckinSuccess(history)` function to jump to activity page.